src/v/cloud_storage/async_manifest_view.h

```diff
@@ -158,6 +160,10 @@ class async_manifest_view {
       std::optional<size_t> size_limit,
       std::optional<std::chrono::milliseconds> time_limit) noexcept;
 
+    const remote_path_provider& path_provider() const {
```

Comment:
This is confusing, why `async_manifest_view` exposes its own path provider which potentially could be different from the one in the `archival_metadata_stm`. If I'm writing the code in the `remote_partition` or `ntp_archiver` I can use `async_manifest_view::path_provider` or `archival_metadata_stm::path_provider`. What is the goal here? Why can't it be routed everywhere in c-tor?

Reply:
Pasting from another PR comment:

I think it makes sense for the remote partition to depend on async_manifest_view's -- the hierarchy at least makes sense because the remote_partition uses the view for both getting segments and cursoring over spillover manifests (done by the view internally with the view's path provider)